src/target/riscv/riscv.h

```diff
@@ -272,9 +272,7 @@ struct riscv_info {
 	struct reg_data_type_union vector_union;
 	struct reg_data_type type_vector;
 
-	/* Set when trigger registers are changed by the user. This indicates we need
-	 * to beware that we may hit a trigger that we didn't realize had been set. */
-	bool manual_hwbp_set;
+	bool *reserved_triggers;
```

Comment:
Just a detail that I noticed - for your consideration:

If you like, the `reserved_triggers` array can be allocated statically:

```suggestion
	bool reserved_triggers[RISCV_MAX_TRIGGERS];
```

The advantages would be:

- Consistency -- the same approach is already used for `trigger_info` in `struct riscv_info`.
- No need for malloc and free - one less thing to worry about and reduced area for coding mistakes.

Reply:
`RISCV_MAX_TRIGGERS == 32`  and `RISCV_MAX_HWBPS == 16`  seem like arbitrary limitations to me.
AFAIK, the specification supports up to `2^XLEN` triggers (limited by `tselect`).
Therefore, I'd like to remove this constants in the future.

Reply:
That's a fair point, so let's leave it as is. (Although I believe that hardware having so many triggers would be rare.)